### ISSUE TITLE
Map field are not currently supported for JSON

### ISSUE THREAD

**Issue by vphantom** [mransan/ocaml-protoc#165]
I'm hitting this error in the `gen_record` functions of `src/compilerlib/pb_codegen_encode_yojson.ml` and `src/compilerlib/pb_codegen_decode_yojson.ml`.

It seems that maps in general, including of scalar types like `map <string,string>` aren't implemented for Yojson I/O yet.

**Comment by mransan** [comment]
Yes this is the line that would need to provide the implementation for instance: https://github.com/mransan/ocaml-protoc/blob/b62a4d5d970533ce99f8c875c3522959ba56a569/src/compilerlib/pb_codegen_encode_yojson.ml#L209
